### PR TITLE
Data + Aria props

### PR DESCRIPTION
**Pull request by thestephenmarshall**
#### Screens

- Adds `data` attributes to some critical areas within kits
- Adds `aria` attributes to Collapsible

#### Breaking Changes

No

#### Runway Ticket URL

https://nitro.powerhrg.com/runway/backlog_items/NUXE-505

#### How to test this

Nothing visual to test. Ensure no accessibility violations in Collapsible kit.

#### Checklist:

- [x] **LABELS** Add a label: `enhancement`, `bug`, `improvement`, `new kit`, `deprecated`, or `breaking`. See [Changelog & Labels](https://github.com/powerhome/playbook/wiki/Changelog-&-Labels) for details.
- [x] **URGENCY** Please select a release milestone
- [x] **DEPLOY** Please add the `Milano` label when you are ready for a review.
- [x] ~~**SCREENSHOT** Please add a screen shot or two.~~
- [x] **SPECS** Please cover your changes with specs.
- [x] **READ DOCS** Please make sure you have read and understand the [Playbook Release Process](https://github.com/powerhome/playbook/wiki/Playbook-Releases)

*The normal release cut off deadline is 3p EDT each week. Please reach out to the release team if you have an urgent request or need a release off cycle.*

### CODE REVIEW

File: playbook/app/pb_kits/playbook/pb_button/_button.jsx [hunk 1]
```diff
@@ -114,26 +114,26 @@ const Button = (props: ButtonPropTypes) => {
...
     <If condition={link !== null}>
       <a
-          {...buttonAria}
-          {...dataProps}
           className={css}
           href={link}
           id={id}
           target={newWindow ? '_blank' : null}
+          {...buttonAria}
+          {...dataProps}
```
Comment (thestephenmarshall) [inline playbook/app/pb_kits/playbook/pb_button/_button.jsx:122]: Spread always goes last

File: playbook/app/pb_kits/playbook/pb_circle_icon_button/_circle_icon_button.jsx [hunk 2]
```diff
@@ -31,6 +32,7 @@ type CircleIconButtonProps = {
 const CircleIconButton = (props: CircleIconButtonProps) => {
   const {
     aria = {},
+    buttonData = {},
```
Comment (thestephenmarshall) [inline playbook/app/pb_kits/playbook/pb_circle_icon_button/_circle_icon_button.jsx:35]: Doesn't break anything already implemented. Now send `buttonData` to add data to the button directly.

File: playbook/app/pb_kits/playbook/pb_collapsible/_collapsible.scss [hunk 3]
```diff
@@ -13,3 +14,8 @@
     height: auto;
   }
 }
+// End: This isn't being used
```
Comment (thestephenmarshall) [inline playbook/app/pb_kits/playbook/pb_collapsible/_collapsible.scss:17]: What is going on with this SCSS?

File: playbook/app/pb_kits/playbook/pb_collapsible/collapsible.html.erb [hunk 4]
```diff
@@ -2,6 +2,7 @@
     aria: object.aria,
     class: object.classname,
     data: object.data,
-    id: object.id) do %>
+    id: object.id,
+    role: "region") do %>
```
Comment (thestephenmarshall) [inline playbook/app/pb_kits/playbook/pb_collapsible/collapsible.html.erb:6]: Adds `role` to the wrapper. https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/Region_role

File: playbook/app/pb_kits/playbook/pb_collapsible/collapsible_content.rb [hunk 5]
```diff
@@ -12,6 +12,10 @@ def data
       def classname
         generate_classname("pb_collapsible_content_kit", "toggle-content", padding, separator: " ")
       end
+
+      def aria_content
+        {labelledby: "collapsible"}.merge! aria
+      end
```
Comment (thestephenmarshall) [inline playbook/app/pb_kits/playbook/pb_collapsible/collapsible_content.rb:18]: Really need to set defaults here incase they aren't added

File: playbook/app/pb_kits/playbook/pb_collapsible/collapsible_main.rb [hunk 6]
```diff
@@ -3,13 +3,20 @@
...
       def data
         Hash(prop(:data)).merge(collapsible_main: true)
       end
 
       def classname
         generate_classname("pb_collapsible_main_kit", padding, separator: " ")
       end
+
+      def aria_main
+        {label: "Toggle Section Content", role: "button"}.merge! aria
```
Comment (thestephenmarshall) [inline playbook/app/pb_kits/playbook/pb_collapsible/collapsible_main.rb:18]: Default aria attributes.

File: playbook/app/pb_kits/playbook/pb_collapsible/collapsible_main.rb [hunk 7]
```diff
@@ -3,13 +3,20 @@
 module Playbook
   module PbCollapsible
     class CollapsibleMain < Playbook::KitBase
+      prop :id, type: Playbook::Props::String,
+                  default: "collapsible"
```
Comment (thestephenmarshall) [inline playbook/app/pb_kits/playbook/pb_collapsible/collapsible_main.rb:7]: If you don't add an id, you get this id

File: playbook/app/pb_kits/playbook/pb_flex/flex.html.erb [hunk 8]
```diff
@@ -1,4 +1,5 @@
 <%= content_tag(:div,
+    aria: object.aria,
```
Comment (thestephenmarshall) [inline playbook/app/pb_kits/playbook/pb_flex/flex.html.erb:2]: Was not implemented!

File: playbook/app/pb_kits/playbook/utilities/numbers.js [hunk 9]
```diff
@@ -0,0 +1,2 @@
+export const randoInt = (multiplier = 100000) => parseInt(Math.random() * multiplier)
```
Comment (thestephenmarshall) [inline playbook/app/pb_kits/playbook/utilities/numbers.js:1]:
```js
console.log(randoInt());
console.log(randoFloat());
```

File: playbook/app/pb_kits/playbook/pb_collapsible/collapsible_content.rb (hunk 5 above)
Comment (bh247484) [reply playbook/app/pb_kits/playbook/pb_collapsible/collapsible_content.rb:18]: Think you should be in the clear here to use `merge!`.  My mistake was merging new values into the `aria` prop's memory location.

You're creating a new hash and then merging `aria` into it.  Which doesn't mutate `aria` prop's memory values, only accesses and copies them into the new hash you declared inline.

File: playbook/app/pb_kits/playbook/pb_collapsible/collapsible_main.rb (hunk 6 above)
Comment (bh247484) [reply playbook/app/pb_kits/playbook/pb_collapsible/collapsible_main.rb:18]: Same approach as mentioned above.

File: playbook/app/pb_kits/playbook/pb_checkbox/_checkbox.jsx [hunk 10]
```diff
@@ -48,15 +49,19 @@ const Checkbox = (props: CheckboxProps) => {
     className
   )
 
+  const childrenWithProps = () => {
+    const cloneChild = (kid) => React.cloneElement(kid, { key: `text-input-${randoInt()}`, ...dataProps })
```
Comment (coleerikson) [inline playbook/app/pb_kits/playbook/pb_checkbox/_checkbox.jsx:53]: just verifying you want `text-input-` here vs. checkbox or something

File: playbook/app/pb_kits/playbook/pb_circle_icon_button/circle_icon_button.html.erb [hunk 11]
```diff
@@ -1,9 +1,8 @@
 <%= content_tag(:div,
   aria: object.aria,
   id: object.id,
-  data: object.data,
   class: object.classname) do %>
-  <%= pb_rails("button", props: {type: object.type, link: object.link, new_window:object.new_window, variant: object.variant, disabled: object.disabled, dark: object.dark}) do %>
+  <%= pb_rails("button", props: {data: object.data, type: object.type, link: object.link, new_window:object.new_window, variant: object.variant, disabled: object.disabled, dark: object.dark}) do %>
```
Comment (coleerikson) [inline playbook/app/pb_kits/playbook/pb_circle_icon_button/circle_icon_button.html.erb:5]: I see `type` was just was missing. Good catch 😬

File: playbook/app/pb_kits/playbook/pb_collapsible/collapsible.html.erb (hunk 4 above)
Comment (coleerikson) [reply playbook/app/pb_kits/playbook/pb_collapsible/collapsible.html.erb:6]: ++

**Review by coleerikson — APPROVED** [review]
LGTM pending some of the convos and questions.

File: playbook/app/pb_kits/playbook/pb_checkbox/_checkbox.jsx (hunk 10 above)
Comment (thestephenmarshall) [inline playbook/app/pb_kits/playbook/pb_checkbox/_checkbox.jsx:53]:
```suggestion
    const cloneChild = (kid) => React.cloneElement(kid, { key: `child-${randoInt()}`, ...dataProps })
```

**Comment by jasoncypret** [comment]
Does this one need alpha testing...looked again and looks more like additions/safe to merge

**Comment by thestephenmarshall** [comment]
Sadly, this one also needs to be re-created off `master`. A new PR will supersede it soon.

**Pull request closed without merging**